llmfoundry/models/hf/hf_causal_lm.py

```diff
@@ -148,3 +152,36 @@ def __init__(self, om_model_config: DictConfig, tokenizer: Tokenizer):
                                           init_device=init_device)
 
         return composer_model
+
+
+class ComposerHFCausalLMFromPython(HuggingFaceModelWithZLoss):
+    """Configures a :class:`.HuggingFaceModel` around a Causal LM that is loaded in memory.
+    Args:
+        model (peft.peft_model.PeftModel or transformers.PreTrainedModel): The HF model loaded into python memory.
+        tokenizer (PreTrainedTokenizer): The tokenizer that the model will use.
+    """
+
+    def __init__(self, model: Union[peft.peft_model.PeftModel,transformers.PreTrainedModel], tokenizer: Tokenizer):
```

Comment:
since the only change here is how the base model is constructed, should we just change the arg to `ComposerHFCausalLM` to accept `Union[DictConfig, peft.peft_model.PeftModel, transformers.PreTrainedModel`?

Reply:
@dakinggg the reason I decided against it is that I didn't want to contaminate the main func with flags that immediately returns a composer model, but happy to change.

Reply:
My opinion is that's it's better to unify them. I think it will just be a simple `if instance(model, DictConfig): do all the extra stuff else: pass through the model`. I think we should refactor (not in this pr) to have a dedicated builder function that takes the config anyway, instead of this class accepting a config as a constructor arg.

Reply:
ok, will make the modification now and test on my end.